File: android/lib/dtos/register_dto.dart
class RegisterDto {
  final String password;
  final String retypePassword;
  final String fullName;
  final String phoneNumber;

  RegisterDto({
    required this.phoneNumber,
    required this.password,
    required this.retypePassword,
    required this.fullName,
  });

  Map<String, dynamic> toJson() {
    return {
      'phoneNumber': phoneNumber,
      'password': password,
      'retypePassword': retypePassword,
      'fullName': fullName,
    };
  }
}

File: admin/src/app/dtos/user/register.dto.ts
import {IsString, IsNotEmpty, IsPhoneNumber, IsDate} from 'class-validator'

export class RegisterDto{
    @IsString()
    @IsNotEmpty()
    password: string;

    @IsString()
    @IsNotEmpty()
    retypePassword: string;

    email:string;

    @IsString()
    fullName:string;

    @IsString()
    address:string;

    @IsPhoneNumber()
    phoneNumber:string;

    @IsDate()
    dateOfBirth:Date;

  constructor(data:any){
    this.phoneNumber=data.phoneNumber;
    this.password=data.password;
    this.retypePassword=data.retypePassword;
    this.fullName=data.fullName;
    this.email=data.email;
    this.address=data.address;
    this.dateOfBirth=data.dateOfBirth;
  }
}